SparkPost/2web2ui
# SA:1664 - Move reportOptions into Context

### What Changed
 - Moved ReportOptions redux state into Context with hooks.
 - Replaced action/reducer/selectors.
 - Replaced instances of reportOptions redux action/reducer with a useReducer.

### How To Test
 - Go to Report Builder and do everything you can think of.
 - Try changing dates, filters, metrics; and check that the UI changes are correct. Also check that the API calls are correct as well.

### To Do
- [ ] Change the patterns? Best Practices?
